sql/planbuilder/builder.go

```diff
@@ -132,7 +157,20 @@ func New(ctx *sql.Context, cat sql.Catalog, es sql.EventScheduler, p sql.Parser)
 		qFlags:         &sql.QueryFlags{},
 		authEnabled:    true,
 		authQueryState: state,
+		overrides:      overrides,
+	}
+}
+
+// NewFromContext creates a new Builder using information from the given context. This will preserve the parser,
+// overrides, etc. as long as the context was created from the server handler. Otherwise, this will use defaults
+// equivalent to calling New with nil options.
+func NewFromContext(ctx *sql.Context, cat sql.Catalog) *Builder {
```

Comment:
You don't need this if the catalog is aware of and can vend information about builder overrides. It's not ideal, but it's not terrible and definitely preferable to jamming this information into the context.